src/charm.py

```diff
@@ -266,9 +272,55 @@ def _on_peer_relation_changed(self, event: RelationChangedEvent) -> None:
             self.unit.status = ActiveStatus()
 
     def _on_config_changed(self, _) -> None:
-        """Handle the config-changed event."""
-        # TODO: placeholder method to implement logic specific to configuration change.
-        pass
+        """Handle configuration changes, like enabling plugins."""
+        if not self.is_cluster_initialised:
+            logger.debug("Early exit on_config_changed: cluster not initialised yet")
+            return
+
+        if not self.unit.is_leader():
+            return
+
+        # Enable and/or disable the extensions.
+        self.enable_disable_extensions()
+
+        # Unblock the charm after extensions are enabled (only if it's blocked due to application
+        # charms requesting extensions).
+        if self.unit.status.message != EXTENSIONS_BLOCKING_MESSAGE:
+            return
+
+        for relation in self.model.relations.get("db", []):
+            if not self.legacy_db_relation.set_up_relation(relation):
+                logger.debug(
+                    "Early exit on_config_changed: db relation requested extensions that are still disabled"
+                )
+                return
+
+        for relation in self.model.relations.get("db-admin", []):
+            if not self.legacy_db_admin_relation.set_up_relation(relation):
+                logger.debug(
+                    "Early exit on_config_changed: db-admin relation requested extensions that are still disabled"
+                )
+                return
```

Comment:
We can loop over the relations in one go
```suggestion
        for relation in [
            *self.model.relations.get("db", []),
            *self.model.relations.get("db-admin", []),
        ]:
            if not self.legacy_db_relation.set_up_relation(relation):
                logger.debug(
                    "Early exit on_config_changed: legacy relation requested extensions that are still disabled"
                )
                return
```

Reply:
You're right. I forgot that the behaviour is the same. I'm gonna update it.